### PR TITLE
Add a test to ensure that `types.d.ts` is valid

### PR DESCRIPTION
**Pull request by markgaze**
## Changes

In #11249, I inadvertently introduced a bug in the types generation, which has since been fixed but I thought I'd add a test to make sure that the output is valid TypeScript (as mentioned in https://github.com/withastro/astro/issues/11274#issuecomment-2174185752). This PR adds that check to the already existing test that checked the types were generated.

I've moved the offending line of code back closer to the `for` loop (but not inside it!) so that it's easier to see that it's related to that and not the JSON schema generation.

## Testing

I moved the line that broke the types generation back into the `for` loop, ran the tests and it failed. When I undid that change and ran the tests again, it passed 🙂 

## Docs

Does not affect the end user so no docs have been written.

### CODE REVIEW

File: packages/astro/test/astro-sync.test.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import assert from 'node:assert/strict';
 import * as fs from 'node:fs';
 import { before, describe, it } from 'node:test';
+import ts from 'typescript';
```
Comment (alexanderniebuhr) [inline packages/astro/test/astro-sync.test.js:4]: I think this is fine, since it is only a test, but I'll let other comment, if we want to import the full `typescript` package here.

**Review by alexanderniebuhr — APPROVED** [review]
@markgaze Thanks for this contribution. No worries on the introduced bug, I didn't catch it in the review either, but very good to have a test for this now. I commented a little question for others to comment on, but otherwise this looks good to me.

File: packages/astro/test/astro-sync.test.js [hunk 2]
```diff
@@ -62,6 +63,20 @@ const createFixture = () => {
 			const expectedPath = new URL(path, astroFixture.config.root).href;
 			assert.equal(writtenFiles[expectedPath].includes(content), true, error);
 		},
+		thenFileShouldBeValidTypescript(path) {
```
Comment (matthewp) [inline packages/astro/test/astro-sync.test.js:66]: I don't think I understand the naming of this. Did you mean `typesFileShouldBeValidTypeScript`?

File: packages/astro/test/astro-sync.test.js (hunk 1 above)
Comment (Princesseuh) [reply packages/astro/test/astro-sync.test.js:4]: This is fine and is the typical way to import from TS

File: packages/astro/test/astro-sync.test.js (hunk 2 above)
Comment (markgaze) [reply packages/astro/test/astro-sync.test.js:66]: I was trying to keep consistency with the functions above 🙂

File: packages/astro/src/content/types-generator.ts [hunk 3]
```diff
@@ -490,6 +490,7 @@ async function writeContentFiles({
 						);
 					}
 				}
+
```
Comment (ematipico) [inline packages/astro/src/content/types-generator.ts:493]: Unrelated change, can you revert?

**Review by ematipico — APPROVED** [review]

File: packages/astro/src/content/types-generator.ts (hunk 3 above)
Comment (markgaze) [reply packages/astro/src/content/types-generator.ts:493]: Will do!
